src/adapters/local.ts

```diff
@@ -12,21 +12,21 @@ import File from '../file';
 import StreamFile from '../stream-file';
 
 export default class LocalAdapter implements IAdapter {
-    private basePath: string;
+    private rootPath: string;
 
-    constructor(opts: { basePath: string }) {
-        this.basePath = opts.basePath;
+    constructor(opts: { rootPath: string }) {
+        this.rootPath = opts.rootPath;
     }
 
-    async create(path: string, contents: string): Promise<IMetadata> {
+    async write(path: string, contents: string): Promise<IMetadata> {
```

Comment:
>And then the adapter only needs to implement `write`.
> -- https://github.com/cohesivelabs/filesystem/issues/7#issue-197297640

Adapters only need to implement `write`, which just writes the file (doesn't care if it exists or not) and the top level code will implement the checks for `create` and `update`.